pennylane_lightning/core/src/simulators/lightning_qubit/bindings/LQubitBindings.hpp

```diff
@@ -162,12 +162,68 @@ void registerControlledGate(PyClass &pyclass) {
  */
 template <class StateVectorT, class PyClass>
 void registerBackendClassSpecificBindings(PyClass &pyclass) {
+    using PrecisionT =
+        typename StateVectorT::PrecisionT; // Statevector's precision
+    using ComplexT = typename StateVectorT::ComplexT;
+    using ParamT = PrecisionT; // Parameter's data precision
+    using np_arr_c = py::array_t<std::complex<ParamT>,
+                                 py::array::c_style | py::array::forcecast>;
+
     registerGatesForStateVector<StateVectorT>(pyclass);
     registerControlledGate<StateVectorT>(pyclass);
-    pyclass.def("applyControlledMatrix", &applyControlledMatrix<StateVectorT>,
-                "Apply controlled operation");
-    pyclass.def("kernel_map", &svKernelMap<StateVectorT>,
-                "Get internal kernels for operations");
+
+    pyclass
+        .def(py::init([](std::size_t num_qubits) {
+            return new StateVectorT(num_qubits);
+        }))
+        .def("resetStateVector", &StateVectorT::resetStateVector)
+        .def(
+            "setBasisState",
+            [](StateVectorT &sv, const size_t index) {
+                sv.setBasisState(index);
+            },
+            "Create Basis State.")
+        .def(
+            "setStateVector",
+            [](StateVectorT &sv, const std::vector<std::size_t> &indices,
+               const np_arr_c &state) {
+                const auto buffer = state.request();
+                std::vector<ComplexT> state_in;
+                if (buffer.size) {
+                    const auto ptr = static_cast<const ComplexT *>(buffer.ptr);
+                    state_in = std::vector<ComplexT>{ptr, ptr + buffer.size};
+                }
+                sv.setStateVector(indices, state_in);
+            },
+            "Set State Vector with values and their corresponding indices")
+        .def(
+            "getState",
```

Comment:
Does the name here match that of LK and LGPU? Same with above and below methods?

Reply:
Similarly, LKokkos and LGPU use `DeviceToHost`.

Reply:
LKokkos call a similar binding, sync_d2h. LGPU calls it syncD2H. Once again the name does not make sense for LQubit.